core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -223,6 +223,8 @@ object NonEmptyVector extends NonEmptyVectorInstances {
     new NonEmptyVector(buf.result)
   }
 
+  def unapply[A](nev: NonEmptyVector[A]): Some[(A, Vector[A])] = Some((nev.head, nev.tail))
```

Comment:
The compiler can check for exhaustiveness if it returns `Some`. It can't when it returns `Option`.